cli/flox/src/commands/init/python.rs

```diff
@@ -0,0 +1,664 @@
+            .filter_map(|provider| match provider {
+                Provide::Found(provider) => Some(provider),
+                _ => None,
+            })
+            .collect::<Vec<_>>();
+
+        fn describe_provider(provider: &dyn Provider) -> String {
+            format!(
+                "* {} ({})\n\n{}",
+                provider.describe_provider(),
+                provider.describe_reason(),
+                textwrap::indent(&provider.describe_customization(), "  ")
+            )
+        }
+
+        message::plain(formatdoc! {"
+            Flox detected a Python project with the following Python provider(s):
+
+            {}
+        ", found_providers.iter().map(|p| describe_provider(p.as_ref())).join("\n")});
+
+        let message = formatdoc! {"
+            Would you like Flox to set up a standard Python environment?
+            You can always change the environment's manifest with 'flox edit'"};
+
+        let accept_options = found_providers
+            .iter()
+            .map(|provider| format!("Yes - with {}", provider.describe_provider()))
+            .collect::<Vec<_>>();
+
+        let n_accept_options = accept_options.len();
+
+        let show_modifications_options = found_providers
+            .iter()
+            .map(|provider| {
+                format!(
+                    "Show suggested modifications for {}",
+                    provider.describe_provider()
+                )
+            })
+            .collect::<Vec<_>>();
+
+        let cancel_option = ["No".to_string()];
+
+        let options = accept_options
+            .iter()
+            .chain(cancel_option.iter())
+            .chain(show_modifications_options.iter())
+            .collect::<Vec<_>>();
+
+        loop {
+            let dialog = Dialog {
+                message: &message,
+                help_message: Some(AUTO_SETUP_HINT),
+                typed: Select {
+                    options: options.clone(),
+                },
+            };
+
+            let (choice, _) = dialog.raw_prompt()?;
+
+            match choice {
+                choice if choice < n_accept_options => {
+                    let _ = self
+                        .selected_provider
+                        .insert(found_providers.swap_remove(choice));
+                    return Ok(true);
+                },
+                c if c == n_accept_options => {
+                    return Ok(false);
+                },
+                choice_with_offset => {
+                    let choice = choice_with_offset - (n_accept_options + 1);
+
+                    let provider = &found_providers[choice];
+                    message::plain(format_customization(&provider.get_init_customization())?);
+                },
+            }
+        }
+    }
+
+    /// Returns the customization of the selected provider or the first found provider
+    ///
+    /// This method assumes that **it is only called if [Self::prompt_user] returned `true`**
```

Comment:
question nonblocking: do you mean if `should_run` returned true? `prompt_user` shouldn't be called if we don't have a tty